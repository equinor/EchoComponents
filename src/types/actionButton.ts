export interface ActionButton {
    title: string;
    onClick: () => void;
    variant: 'contained' | 'outlined' | 'ghost' | 'ghost_icon' | undefined;
}
