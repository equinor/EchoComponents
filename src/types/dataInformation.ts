export interface DataInformation {
    numberOfItems: number;
    label: string;
    ariaLabel?: string;
    onTagInfoClicked: () => void;
    itemType: string;
    itemButtonClassName?: string;
}
