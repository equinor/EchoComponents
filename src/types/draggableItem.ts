export interface DraggableItem {
    id: number;
    element: React.ReactNode;
}
