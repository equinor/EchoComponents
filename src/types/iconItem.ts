export interface IconItem {
    icon: string;
    onClick: () => void;
}
