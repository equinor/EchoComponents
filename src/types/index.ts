export * from './actionButton';
export * from './dataInformation';
export * from './draggableItem';
export * from './expandableRowProps';
export * from './iconItem';
export * from './iconListItem';
export * from './linkCardItem';
