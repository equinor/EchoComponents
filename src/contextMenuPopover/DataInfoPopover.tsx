import { DotProgress } from '@equinor/eds-core-react';
import React from 'react';
import { DataInformation } from '../types/dataInformation';
import DataInfoButton from './DataInfoButton';
import style from './dataInfoPopover.module.css';

export interface DataInfoPopoverProps {
    dataToShow: DataInformation[];
    isLoading: boolean;
}

export const DataInfoPopover: React.FC<DataInfoPopoverProps> = ({ dataToShow, isLoading }: DataInfoPopoverProps) => {
    const getDataGroupings = dataToShow.reduce((dictionary: { [itemType: string]: DataInformation[] }, current) => {
        if (!dictionary[current.itemType]) {
            dictionary[current.itemType] = [current];
        } else {
            dictionary[current.itemType] = [...dictionary[current.itemType], current];
        }
        return dictionary;
    }, {});

    return (
        <div className={style.wrapper}>
            <div className={style.arrow}></div>
            <div className={style.optionsPopover}>
                {isLoading && dataToShow.length === 0 && <DotProgress variant="white" />}
                {Object.keys(getDataGroupings).map((key) => {
                    return (
                        <div key={key} className={style.groupWrapper}>
                            {getDataGroupings[key].map((data, index) => {
                                return <DataInfoButton data={data} key={index} />;
                            })}
                        </div>
                    );
                })}
            </div>
        </div>
    );
};

export default DataInfoPopover;
