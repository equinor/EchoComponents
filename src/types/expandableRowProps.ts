import { IconItem } from './iconItem';

export interface ExpandableRowProps {
    iconItems: IconItem[];
    expanded: boolean;
    setExpanded: (status: boolean) => void;
}
