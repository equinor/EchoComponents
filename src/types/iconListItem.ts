import { IconItem } from './iconItem';

export interface IconListItem {
    title?: string;
    subTitle?: string;
    icons: IconItem[];
}
