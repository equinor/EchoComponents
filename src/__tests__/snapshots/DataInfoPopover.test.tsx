import DataInfoPopover from '../../components/contextMenuPopover/DataInfoPopover';
import { DataInformation } from '../../types/dataInformation';
import React from 'react';
import renderer from 'react-test-renderer';

it('renders DataInfoPopover without data correctly', () => {
    const dataToShow: DataInformation[] = [];
    const isLoading = false;

    const tree = renderer.create(<DataInfoPopover dataToShow={dataToShow} isLoading={isLoading} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders DataInfoPopover with data correctly', () => {
    const dataToShow: DataInformation[] = [
        {
            numberOfItems: 5,
            label: 'P&ID',
            ariaLabel: 'P&ID',
            onTagInfoClicked: jest.fn(),
            itemType: 'document'
        },
        {
            numberOfItems: 1,
            label: 'D&ID',
            onTagInfoClicked: jest.fn(),
            itemType: 'document'
        }
    ];
    const isLoading = false;

    const tree = renderer.create(<DataInfoPopover dataToShow={dataToShow} isLoading={isLoading} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders DataInfoPopover with multiple item types correctly', () => {
    const dataToShow: DataInformation[] = [
        {
            numberOfItems: 5,
            label: 'P&ID',
            ariaLabel: 'P&ID',
            onTagInfoClicked: jest.fn(),
            itemType: 'document'
        },
        {
            numberOfItems: 1,
            label: 'D&ID',
            onTagInfoClicked: jest.fn(),
            itemType: 'document'
        },
        {
            numberOfItems: 5,
            label: 'M1',
            onTagInfoClicked: jest.fn(),
            itemType: 'notifications'
        },
        {
            numberOfItems: 1,
            label: 'M2',
            onTagInfoClicked: jest.fn(),
            itemType: 'notifications'
        }
    ];
    const isLoading = false;

    const tree = renderer.create(<DataInfoPopover dataToShow={dataToShow} isLoading={isLoading} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders DataInfoPopover with loading indicator', () => {
    const dataToShow: DataInformation[] = [];
    const isLoading = true;

    const tree = renderer.create(<DataInfoPopover dataToShow={dataToShow} isLoading={isLoading} />).toJSON();
    expect(tree).toMatchSnapshot();
});
