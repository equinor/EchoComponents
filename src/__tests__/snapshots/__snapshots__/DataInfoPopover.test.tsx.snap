// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DataInfoPopover with data correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="arrow"
  />
  <div
    className="optionsPopover"
  >
    <div
      className="groupWrapper"
    >
      <button
        aria-label="P&ID"
        className="sc-gsTCUz eUnAE undefined button buttonWithBadge"
        data-count={5}
        disabled={false}
        onClick={[MockFunction]}
        tabIndex={0}
        type="button"
      >
        <span
          className="sc-bdfBwQ ibGDsh"
        >
          P&ID
        </span>
      </button>
      <button
        aria-label="D&ID"
        className="sc-gsTCUz eUnAE undefined button "
        data-count={1}
        disabled={false}
        onClick={[MockFunction]}
        tabIndex={0}
        type="button"
      >
        <span
          className="sc-bdfBwQ ibGDsh"
        >
          D&ID
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders DataInfoPopover with loading indicator 1`] = `
<div
  className="wrapper"
>
  <div
    className="arrow"
  />
  <div
    className="optionsPopover"
  >
    <svg
      className="sc-fWPcDo gbIjsj  white-progress"
      color="#fff"
      height="8px"
      role="progressbar"
      viewBox="0 0 16 4"
      width="32px"
    >
      <circle
        cx={2}
        cy={2}
        r={2}
      />
      <circle
        cx={8}
        cy={2}
        r={2}
      />
      <circle
        cx={14}
        cy={2}
        r={2}
      />
    </svg>
  </div>
</div>
`;

exports[`renders DataInfoPopover with multiple item types correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="arrow"
  />
  <div
    className="optionsPopover"
  >
    <div
      className="groupWrapper"
    >
      <button
        aria-label="P&ID"
        className="sc-gsTCUz eUnAE undefined button buttonWithBadge"
        data-count={5}
        disabled={false}
        onClick={[MockFunction]}
        tabIndex={0}
        type="button"
      >
        <span
          className="sc-bdfBwQ ibGDsh"
        >
          P&ID
        </span>
      </button>
      <button
        aria-label="D&ID"
        className="sc-gsTCUz eUnAE undefined button "
        data-count={1}
        disabled={false}
        onClick={[MockFunction]}
        tabIndex={0}
        type="button"
      >
        <span
          className="sc-bdfBwQ ibGDsh"
        >
          D&ID
        </span>
      </button>
    </div>
    <div
      className="groupWrapper"
    >
      <button
        aria-label="M1"
        className="sc-gsTCUz eUnAE undefined button buttonWithBadge"
        data-count={5}
        disabled={false}
        onClick={[MockFunction]}
        tabIndex={0}
        type="button"
      >
        <span
          className="sc-bdfBwQ ibGDsh"
        >
          M1
        </span>
      </button>
      <button
        aria-label="M2"
        className="sc-gsTCUz eUnAE undefined button "
        data-count={1}
        disabled={false}
        onClick={[MockFunction]}
        tabIndex={0}
        type="button"
      >
        <span
          className="sc-bdfBwQ ibGDsh"
        >
          M2
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders DataInfoPopover without data correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="arrow"
  />
  <div
    className="optionsPopover"
  />
</div>
`;
